### ISSUE TITLE
Does the model need accumulate the gradients?

### ISSUE THREAD

**Issue by aflyingnoob** [salesforce/EntropyPenalty#1]
At line 220 in [main.py](https://github.com/salesforce/corr_based_prediction/blob/master/main.py), the zero_grad is outside the training stage, the gradients will be accumulated along the training data.
